### PR TITLE
Multiple answer validation

### PR DESCRIPTION
**Pull request by tgandrews**
### What is the context of this PR?
This adds the ability to perform validation against the sum of answers on a single page.

### How to review 
1. Ensure this works as expected

### CODE REVIEW

**Comment by tgandrews** [comment]
[Multiple answer validation - totals (6)](https://trello.com/c/pkIcjKHs/408-multiple-answer-validation-totals-6)

File: eq-author/src/components/Forms/Number/index.js [hunk 1]
```diff
@@ -40,86 +40,77 @@ const UnitSymbol = styled.div`
...
-    if (this.props.onBlur) {
-      setImmediate(() => {
-        this.props.onBlur();
-      });
+  const handleBlur = () => {
+    if (value === null) {
+      onChange({ name, value: props.default });
+    }
+    if (onBlur) {
+      setImmediate(onBlur);
```
[the remark this answers is not in the document]
Comment (tgandrews) [reply eq-author/src/components/Forms/Number/index.js]: Resolved. `setTimeout` gets the time of 0 by default so no need to specify it.

File: eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  return (
+    <Margin>
+      <Grid>
+        <Column cols={5}>
+          <Flex>
+            <ValidationTitle style={{ margin: "0 1em 0 0" }}>
+              Total
+            </ValidationTitle>
+
+            <EqualsSelect
```
Comment (hamishtaplin) [inline eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js]: This needs a label (albeit visually hidden)

File: eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+const EqualsSelect = styled(Select)`
+  width: 8em;
+`;
+
+const Pills = styled(ValidationPills)`
+  margin-top: -2em;
+`;
+
+const Margin = styled.div`
```
Comment (hamishtaplin) [inline eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js]: Remove the margin, it eats up too much space.

File: eq-author/src/App/page/Design/Validation/GroupValidations/index.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    return (
+      <>
+        <TotalButton
+          onClick={this.handleButtonClick}
+          data-test="sidebar-button-total-value"
+          disabled={!total}
+        >
+          <TotalIcon />
+          <Details>
+            <Title>Total</Title>
```
Comment (hamishtaplin) [inline eq-author/src/App/page/Design/Validation/GroupValidations/index.js]: This should display the "comparator". Eg. "Total less than or equal to"

File: eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  total,
+  type,
+  onChange,
+  onUpdate,
+  onChangeUpdate,
+}) => {
+  return (
+    <Margin>
+      <Grid>
+        <Column cols={5}>
```
Comment (hamishtaplin) [inline eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js]: Change both columns to `6`

File: eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  margin-left: auto;
+`;
+
+const Flex = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: flex-end;
+`;
+
+const EqualsSelect = styled(Select)`
```
Comment (hamishtaplin) [inline eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js]: Remove this component and use the raw `Select` with auto width.

File: eq-author/src/App/page/Design/Validation/GroupValidations/index.js (hunk 4 above)
Comment (tgandrews) [reply eq-author/src/App/page/Design/Validation/GroupValidations/index.js]: Would the symbol be enough?

File: eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+import { Select } from "components/Forms";
+import { Grid, Column } from "components/Grid";
+import withEntityEditor from "components/withEntityEditor";
+
+import withChangeUpdate from "enhancers/withChangeUpdate";
+import withPropRemapped from "enhancers/withPropRemapped";
+
+import totalFragment from "graphql/fragments/total-validation-rule.graphql";
+
+import { ValidationPills } from "../../../ValidationPills";
```
[the remark this answers is not in the document]
Comment (tgandrews) [reply eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js]: As discussed was trying to encapsulate the shared validations logic within the validations folder and having an absolute path would couple this component to the location of the validation component.

File: eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js (hunk 6 above)
Comment (tgandrews) [reply eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js]: Done

File: eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js (hunk 5 above)
Comment (tgandrews) [reply eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js]: Done

File: eq-author/src/App/page/Design/Validation/GroupValidations/index.js (hunk 4 above)
Comment (tgandrews) [reply eq-author/src/App/page/Design/Validation/GroupValidations/index.js]: I have added the symbol in a bracket

File: eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js (hunk 3 above)
Comment (tgandrews) [reply eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js]: Done

File: eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js (hunk 2 above)
Comment (tgandrews) [reply eq-author/src/App/page/Design/Validation/GroupValidations/TotalValidation/TotalValidationEditor/index.js]: Done

**Pull request merged**